enterprise/server/remote_execution/executor/executor.go

```diff
@@ -243,16 +238,12 @@ func (s *Executor) ExecuteTaskAndStreamResults(ctx context.Context, st *repb.Sch
 		return true, err
 	}
 	md.ExecutionStartTimestamp = timestamppb.Now()
-	maxDuration := infiniteDuration
-	if currentDeadline, ok := ctx.Deadline(); ok {
-		maxDuration = time.Until(currentDeadline)
-	}
```

Comment:
This block was not needed because the `WithTimeout` on line 255 below should already be respecting any existing deadline in the context.